3rdParty/fuerte/src/HttpConnection.cpp

```diff
@@ -399,8 +399,8 @@ void HttpConnection<ST>::asyncWriteNextRequest() {
   
   auto self = shared_from_this();
   asio_ns::async_write(_protocol.socket, buffers,
-                       [this, self, item](asio_ns::error_code const& ec,
-                                          std::size_t transferred) {
+                       [this, self, item = std::move(item)](asio_ns::error_code const& ec,
+                                                            std::size_t transferred) mutable {
     _bytesToSend.fetch_sub(item->_request->payloadSize(), std::memory_order_release);
     asyncWriteCallback(ec, transferred, std::move(item));
```

Comment:
`item` is moved-from twice here. It is probably better to move back to a shared_ptr like before, especially since the previously assumed leak can not occur (see my comment https://github.com/arangodb/arangodb/pull/7962#discussion_r254174077)

Reply:
I don't think it matters much, I mean the cost of moving a unique_ptr is probably nil

Reply:
The problem is that `item` is null after the first move, i.e., the call to `asyncWriteCallback` always passes an empty unique_ptr.

Reply:
`item` inside the lambda is different from the outisde `item`

Reply:
My bad! I misinterpreted the diff and thought the `asyncWriteCallback` was outside the lambda. All good then! :)